src/routes/SignUp.jsx

```diff
@@ -0,0 +1,50 @@
+import { getAuth, updateEmail, createUserWithEmailAndPassword } from 'firebase/auth';
+import React, { useState } from 'react';
+import { auth } from '../utils/firebase';
+
+const SignUp = () => {
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const auth = getAuth();
+
+  createUserWithEmailAndPassword(auth, email, password)
+    .then((userCredential) => {
+      console.log(userCredential);
+      setEmail('');
+      setPassword('');
+    })
+    .catch((error) => {
+      console.log(error);
+    });
+
+  const signUp = (e) => {
+    e.preventDefault();
+    updateEmail(auth.currentUser, 'user@example.com')
```

Comment:
why are we doing this? we wont need this if we move createUserWithEmailAndPassword here

Reply:
basically we are updating email that doens't exist yet.

Reply:
You might thought that this works because  createUserWithEmailAndPassword is called everytime when the component get rerendered and updateEmail is called when the signup button is clicked